docs/vocabulary.md

```diff
@@ -0,0 +1,312 @@
+- **Trigger**: A criteria for triggering a new Pipeline run or Build.
+
+### Jenkins X
+
+[Jenkins X](https://jenkins-x.io/) provides pipeline automation, built-in
+GitOps, and preview environments to help teams collaborate and accelerate
+their software delivery at any scale.
+
+Some of the core Jenkins X terms are listed below. [[5]]
+
+- **Step**: Basic building block to execute one or more commands.
+- **Stage**: Collection of steps and the required configuration.
+- **Pipeline**: Collection of one or more stages to run.
+- **Agent**: Container image or Jenkins agent to schedule pipelines on.
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline.
+
+### Screwdriver
+
+[Screwdriver](https://screwdriver.cd) is an open source build platform designed
+for Continuous Delivery.
+
+Some of the core Screwdriver terms are listed below. [[6]]
+
+- **Step**: List of commands to execute.
+- **Job**: Collection of the steps to run.
+- **Workflow**: Collection and order of jobs to run.
+- **Pipeline**: Used interchangeably with workflow. (*citation needed*)
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline
+or job. (*citation needed*)
+
+### Spinnaker
+
+[Spinnaker](https://www.spinnaker.io/) is an open source, multi-cloud continuous
+delivery platform for releasing software changes with high velocity and
+confidence.
+
+Some of the core Spinnaker terms are listed below. [[7]]
+
+- **Task**: An atomic function to perform.
+- **Stage**: A collection of sequential Tasks and composed Stages that describe
+a higher-level action the Pipeline will perform either linearly or in parallel.
+- **Pipeline**: The pipeline is the key deployment management construct in
+Spinnaker. It consists of a sequence of actions, known as stages.
+- **Pipeline Template**: A parameterized pipeline, minus the pipeline
+configuration found on a pipeline instance. This template helps developers
+create pipelines that follow a pattern that you establish.
+- **Pipeline configuration**: The same as the configuration for a pipeline
+not created from a template, plus variable bindings and a reference to the
+template.
+- **Cluster**: Logical groupings of Server Groups in Spinnaker.
+- **Provider**: A Cloud Provider is an interface to a set of virtual resources
+that Spinnaker has control over.
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline.
+
+### Tekton
+
+[The Tekton Pipelines](https://github.com/tektoncd/pipeline) project provides
+k8s-style resources for declaring CI/CD-style pipelines.
+
+Some of the core Tekton terms are listed below. [[8]]
+
+- **Step**: a specific function to perform.
+- **Task**: is a collection of sequential steps you would want to run as part of your
+continuous integration flow. A task will run inside a pod on your cluster.
+- **ClusterTask**: Similar to Task, but with a cluster scope.
+- **Pipeline**: stateless, reusable, paramterized collection of tasks. Tasks linked
```

Comment:
I think there is a typo where line 185 is meant to continue this line - i'd replace this with something like:

Pipeline: a reusable parameterized directed acyclic graph of Tasks. Task inputs and outputs can be linked by string values (results) and files (workspaces)

(I'm removing PipelineResources from this on purpose since the future of PipelineResources is not clear)

Reply:
ditto